services/volumesrv/store.go

```diff
@@ -11,21 +17,85 @@ func NewPostgresStore() Store {
 }
 
 func (p postgresStore) VolumeAdd(projectID string, volume types.Volume) error {
-	//TODO implement me
-	panic("implement me")
+	ctx := context.Background()
+	_, err := db.Q.VolumeCreate(ctx, db.VolumeCreateParams{
+		ID:        volume.ID,
+		ProjectID: projectID,
+		Provider:  volume.Provider.String(),
+	})
+	if err != nil {
+		return &types.Error{
+			Code:    http.StatusBadRequest,
+			Message: "Volume could not be created",
+			Err:     err,
+		}
+	}
+
+	return nil
+}
+
+func (p postgresStore) VolumeList(projectID string) ([]types.Volume, error) {
+	vols, err := db.Q.VolumeList(context.Background(), projectID)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("failed to get volumes from db: %w", err)
+	}
+
+	out := make([]types.Volume, 0, len(vols))
+	for _, v := range vols {
+		out = append(out, volumeFromDB(v))
+	}
+
+	return out, nil
 }
 
 func (p postgresStore) VolumeGet(projectID, idOrName string) (types.Volume, error) {
-	//TODO implement me
-	panic("implement me")
+	vol, err := db.Q.VolumeGet(context.Background(), db.VolumeGetParams{
+		ProjectID: projectID,
+		ID:        idOrName,
+	})
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return types.Volume{}, &types.Error{
+				Code:    http.StatusBadRequest,
+				Message: "Volume not found",
+				Err:     err,
+			}
+		}
+		return types.Volume{}, fmt.Errorf("failed to get volume from db: %w", err)
+	}
+
+	return volumeFromDB(vol), nil
 }
 
-func (p postgresStore) VolumeDelete(id string) {
-	//TODO implement me
-	panic("implement me")
+func (p postgresStore) VolumeDelete(id string) error {
+	ctx := context.Background()
+	if err := db.Q.VolumeDelete(ctx, id); err != nil {
+		return &types.Error{
+			Code:    http.StatusBadRequest,
+			Message: "Volume could not be deleted",
+			Err:     err,
+		}
+	}
```

Comment:
This is not the correct error. This needs to be parsed for `sql.NoRows`. If that'at not the issue, you should return a simple wrapped error. The API will then handle throwing a `InternalServerError`